pkg/github/github.go

```diff
@@ -322,6 +326,30 @@ func (g *GitHubClient) ListIssueComments(ctx context.Context, owner, repo string
 	return &IssueCommentResponse{Comments: comments, Pagination: pagination}, nil
 }
 
+// RepoUserPermissionLevel gets the repository permission level for a user. The possible permissions values
+// are admin, write, read, none.
+func (g *GitHubClient) RepoUserPermissionLevel(ctx context.Context, owner, repo, user string) (string, error) {
+	var permissionLevel string
+
+	if err := g.withRetries(ctx, func(ctx context.Context) error {
+		ghPermissionLevel, resp, err := g.client.Repositories.GetPermissionLevel(ctx, owner, repo, user)
```

Comment:
Will this include if they have permissions at the Org or Enterprise level?

Reply:
If the permissions are inherited, I believe the answer is yes.

https://docs.github.com/en/free-pro-team@latest/rest/collaborators/collaborators?apiVersion=2022-11-28#get-repository-permissions-for-a-user